ephios/plugins/questionnaires/signals.py

```diff
@@ -174,3 +178,30 @@ def save_signup(
                 SavedAnswer.objects.filter(
                     user=participant.user, question_id=Question.get_pk_from_slug(name)
                 ).delete()
+
+
+@receiver(
+    insert_html,
+    sender=HTML_DISPOSITION_PARTICIPATION,
+    dispatch_uid="ephios.plugins.questionnaires.signals.insert_html",
+)
+def disposition_render_responses(request, participation: AbstractParticipation, **kwargs):
+    answers = participation.answer_set.all()
+
+    if not answers:
+        return ""
+
+    return render_to_string(
+        "questionnaires/disposition_participation_answers.html", {"answers": answers}
+    )
+
+
+@receiver(shift_action, dispatch_uid="ephios.plugins.questionnaires.signals.shift_action")
+def add_shift_action(sender, shift: Shift, request, **kwargs):
+    if hasattr(shift, "questionnaire") and shift.questionnaire and shift.questionnaire.questions:
+        return [
+            {
+                "label": "View answers",
```

Comment:
is it clear for the user what "answers" is referring to? But mentioning questionnaires here would probably lead to a button that is too big